modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/NameClashModelValidator.java

```diff
@@ -199,12 +220,34 @@ private void validateNameClashes(Collection<? extends Named>... collections) {
             i++;
           }
 
-          errorMessage.append(format(" which name is '%s'. Names should be unique", entry.getKey()));
+          errorMessage.append(format(" which alias name is '%s'. Names should be unique", entry.getKey()));
           throw new IllegalModelDefinitionException(errorMessage.toString());
         }
       });
     }
 
+    private void validateSingularizeNameClashesWithTopLevels() {
+      Map<String, Collection<TopLevelParameter>> singularClashes = topLevelParameters.keySet().stream()
```

Comment:
this would be much more efficient and clear if you iterate over the entry set instead of the key set